### ISSUE TITLE
Serializing a named list unintuitive

### ISSUE THREAD

**Issue by msmith-techempower** [FasterXML/jackson-dataformat-xml#159]
My use case is very simple, I have to create an XML document that has, among others, a list of "item" nodes with a parent "items" node. It could look like this:

```
<channel>
  <items>
    <item>...</item>
    <item>...</item>
  </items>
</chanell>
```

However, the naive approach of using a list/array cannot seem to produce this with any amount of configuration via annotations:
```java
public class Channel
{
  public final List<Item> items = new ArrayList<>();
}

public class Item
{
}

// Produces: <channel><items><items></items>...</items></channel>
```

Ultimately, the only way I was able to create the output conforming to the schema from above was to create a useless wrapper class, turn off wrapping, and name my list nonsensically:
```java
public class Channel
{
  public final Items items = new Items();
}

public class Items
{
  @JacksonXmlElementWrapper(useWrapping=false)
  public final List<Item> item = new ArrayList<>();
}

public class Item
{
}

// Produces: <channel><items><item></item>...</items></channel>
```

This does result in the correct output schema, but now I am forced to access my objects in a non-intuitive way. For instance, if I want to add an item to my list of items, I must call the following:
```java
channel.items.item.add(new Item());
```
which looks like I am adding an Item to another Item, rather than the list of items.

What I expected to work but did not was the following:
```java
public class Channel
{
  // with wrapping enabled, the parent node should be "<items>"
  public final List<Item> items = new ArrayList<>();
}

// Each Item instance should be rendered as "<item>"
@JacksonXmlRootElement(localName="item")
public class Item
{
}
```
This would allow me to access my data in a sane manner:
```java
channel.items.add(new Item());
```

TL;DR - It seems to me that `@JacksonXmlRootElement(localName="item")` does not work on any object in a collection/array, and instead the localName of the collection is used for the parent node (correct) AND each object in the collection/array (incorrect).

**Comment by jacobharris919** [comment]
Have you tried this?

```
   public class Channel
   {
      @JacksonXmlElementWrapper(localName = "items")
      @JacksonXmlProperty(localName = "item")
      public final List<Item> items = new ArrayList<Item>();
    }
```

**Comment by jackygurui** [comment]
I agree with @msmith-techempower that the ```@JacksonXmlRootElement(localName="item")``` should be honoured for each object in the collection.

I am experiencing a similar issue, and instead of posting another example, please allow me reuse the example posted by @msmith-techempower with a little twist for the purpose of illustration:

When you have a collection like this:
```xml
<channel>
  <items>
    <item>...</item>
    <item>...</item>
  </items>
</channel>
```

Quite often that  ```Item``` is an abstract type, it could actually be ```Pen```,  ```Pencil``` or ```Paper``` type, thus naturally, one might expect the output would be like this:
```xml
<channel>
  <items>
    <pen>...</pen>
    <pencil>...</pencil>
    <paper>...</paper>
  </items>
</channel>
```

PS: Each type could have different namespaces.

**Comment by jackygurui** [comment]
OK, I might have found a solution:
```java
public class Channel
   {
      @JacksonXmlElementWrapper(useWrapping = false)
      @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.WRAPPER_OBJECT)
      public final List<Item> items = new ArrayList<Item>();
    }
```

**Comment by cowtowncoder** [comment]
@jackygurui Yes, you are correct with the solution. Another, perhaps more common way, would be to annotate `Item` class, but both approaches work.

As to wrapping/unwrapping It is not a bug, as Jackson defaults to "wrapped" serialization. But behavior is different than what JAXB would do, which can be non-intuitive.
There is a way to change the default for xml module if you want to default to unwrapped notation.

**Comment by cowtowncoder** [comment]
I will close this issue because I am not 100% if there is work to do here. It is true that JAXB defaults (no wrapping for `List`s, by default) and Jackson defaults (wrapping is enabled by default) differ, but at this point there isn't much that can be done (at least until Jackson 3). There are also other issue reports that I suspect cover other parts of the problem.

If parts of problems included here still occur, and there is no other issue, please file a new issue: that way it is easier to exactly what is still missing or not working correctly. So just closing to try to clean up the issue list.
